hydro_serving_writer/buffer.py

```diff
@@ -0,0 +1,163 @@
+            buffer = ModelVersionBufferPool.get_buffer(model_version)
+            buffer.write(request.ByteSize(), request.SerializeToString())
+
+    @staticmethod
+    def group_request(request: writer.WriteManyRequest) -> Dict[int, writer.WriteManyRequest]:
+        group = defaultdict(writer.WriteManyRequest)
+        for row in zip(*(getattr(request, name) for name in scheme.names)):
+            group[row[0]].MergeFrom(
+                writer.WriteManyRequest(
+                    **dict(zip(scheme.names, map(lambda x: [x], row)))
+                )
+            )
+        return group
+        
+
+class ModelVersionBuffer:
+    def __init__(self, model_version, buffer_size=1024*1024*16, *args, **kwargs):
+        self.__version = model_version
+        self.__buffer_size = buffer_size
+        self._buffer = None
+        self._output_stream = None
+        logger.debug("Initialized new {}".format(self))
+
+    def __repr__(self):
+        return f"ModelVersionBuffer(version={self.__version})"
+    
+    @property
+    def output_stream(self) -> pa.BufferOutputStream: 
+        if not hasattr(self, '_buffer') or self._buffer is None:
+            self._allocate_buffer()
+        if not hasattr(self, '_output_stream') or self._output_stream is None:
+            self._output_stream = pa.output_stream(self._buffer)
+        return self._output_stream
+
+    def _allocate_buffer(self, size=None):
+        self._buffer = pa.allocate_buffer(size or self.__buffer_size)
+        logger.debug("Allocated new buffer of size {}".format(self.__buffer_size))
+        
+    def _request_can_be_written(self, length: int) -> bool:
+        if length > self.__buffer_size:
+            raise ValueError("Message is greater than buffer size")
+        return self.output_stream.tell() + length + 4 < self.__buffer_size
+
+    def _read_request_length(self, stream: pa.BufferReader) -> int: 
+        if stream.tell() + 4 <= self.__buffer_size: 
+            return int.from_bytes(stream.read(4), "big")
+    
+    def read(self) -> writer.WriteManyRequest:
+        """ Parse binary data from buffer into ProtoBuf message """
+        
+        stream = pa.input_stream(self._buffer)
+        requests = writer.WriteManyRequest()
+        length = self._read_request_length(stream)
+        while length and length < self.__buffer_size - stream.tell():
+            requests.MergeFromString(stream.read(length))
+            length = self._read_request_length(stream)
+        logger.debug("Read buffer")
+        return requests
+
+    def write(self, length: int, body: bytes):
+        """ Write request into buffer """
+
+        if not self._request_can_be_written(length):
+            requests = self.flush()
+
+            timestamp = int(datetime.datetime.utcnow().timestamp())
+            source_path = "{}.parquet".format(timestamp)
+            destination_path = "dump/model_version={}/timestamp={}.parquet".format(
```

Comment:
```suggestion
            destination_path = f"dump/model_version={self.__version}/timestamp={timestamp}.parquet"
```

Reply:
I think, it's better to stick with a single code-style and use `format` everywhere

Reply:
This is decreases the load of constructing log messages, when it's not required.

Reply:
Okay, I've just shown that it might be better to use f-strings instead of `format`, as it is the first implementation of the project (though, code style might be inherited from previous hydro projects).

Also, f-strings are a bit faster: https://realpython.com/python-f-strings/#speed

Reply:
Yeah, f-strings are faster but I'm just worried about this sort of issue: https://stackoverflow.com/questions/54367975/python-3-7-logging-f-strings-vs